src/sentry/filters/legacy_browsers.py

```diff
@@ -11,14 +14,53 @@
     'Safari': 6,
     'Edge': 0,
     'Opera': 15,
+    'Android': 4,
 }
 
 
+class LegacyBrowserFilterSerializer(serializers.Serializer):
+    # TODO: maybe find a serializer that handles multple values already
+    active = serializers.BooleanField()
+    subfilters = MultipleChoiceField(choices=[
+        'ie8',
+        'ie9',
+        'opera',
+        'android',
+        'safari'
+    ])
+
+
 class LegacyBrowsersFilter(Filter):
     id = 'legacy-browsers'
     name = 'Filter out known errors from legacy browsers'
     description = 'Older browsers often give less accurate information, and while they may report valid issues, the context to understand them is incorrect or missing.'
     default = False
+    serializer_cls = LegacyBrowserFilterSerializer
+
+    def is_enabled(self):
+        # May be either a '1' or an iterable for new style
+        return ProjectOption.objects.get_value(
+            project=self.project,
+            key='filters:{}'.format(self.id),
+            default='1' if self.default else '0',
+        )
+
+    def enable(self, value=None):
+        if value is None:
+            value = {}
+
+        option_val = '0'
+        if 'active' in value:
+            if value['active']:
+                option_val = '1'
+        elif 'subfilters' in value and len(value['subfilters']) > 0:
+            option_val = set(value['subfilters'])
```

Comment:
We should make sure we're only allowing a whitelist of `subfilter` values here. If we're assuming that this _had_ to get sanitized and validated through the form, I'd comment that we're explicitly choosing not to do validation here because of that since it's not 100% obvious.